src/epics/handleBetaModeToggle.js

```diff
@@ -1,38 +1,92 @@
+import gql from 'graphql-tag'
 import { Observable } from 'rxjs'
 import {
   USER_TOGGLE_BETA_MODE,
   APP_USER_BETA_MODE_SAVE,
-  APP_USER_NEWS_SAVE
+  APP_USER_NEWS_SAVE,
+  APP_USER_BETA_MODE_SAVE_ERROR,
+  CHANGE_USER_DATA
 } from './../actions/types'
-import { saveKeyState } from '../utils/localStorage'
+import { saveKeyState, loadKeyState } from '../utils/localStorage'
+import { handleErrorAndTriggerAction } from './utils'
 
-const handleBetaModeToggle = (action$, store) =>
+const BETA_MODE_MUTATION = gql`
+  mutation updateUserSettings($isBetaMode: Boolean!) {
+    updateUserSettings(settings: { isBetaMode: $isBetaMode }) {
+      isBetaMode
+    }
+  }
+`
+
+const handleBetaModeToggle = (action$, store, { client }) =>
   action$
     .ofType(USER_TOGGLE_BETA_MODE)
-    .debounceTime(200)
     .map(() => {
       const isBetaModeEnabled = !store.getState().rootUi.isBetaModeEnabled
-      saveKeyState('isBetaModeEnabled', isBetaModeEnabled)
+      saveKeyState('isBetaMode', isBetaModeEnabled)
       if (!isBetaModeEnabled) saveKeyState('isNewsEnabled', false)
       return Observable.of(isBetaModeEnabled)
     })
     .mergeMap(({ value }) => {
-      return value
-        ? Observable.of({
-          type: APP_USER_BETA_MODE_SAVE,
-          payload: value
+      const { data } = store.getState().user
+      if (data && data.id) {
+        const mutation = client.mutate({
+          mutation: BETA_MODE_MUTATION,
+          variables: { isBetaMode: value }
         })
-        : Observable.from([
-          {
+        return Observable.from(mutation)
+          .mergeMap(({ data: { updateUserSettings: { isBetaMode } } }) => {
+            return value
+              ? Observable.of({ type: APP_USER_BETA_MODE_SAVE, payload: true })
+              : Observable.from([
+                { type: APP_USER_BETA_MODE_SAVE, payload: false },
+                { type: APP_USER_NEWS_SAVE, payload: false }
+              ])
+          })
+          .catch(handleErrorAndTriggerAction(APP_USER_BETA_MODE_SAVE_ERROR))
+      } else {
```

Comment:
`else` is not necessary, because of the [`return` statement inside `if` block](https://github.com/santiment/app/pull/457/files#diff-0e20b76b28777795df84b9091165cf10R37) (`LOC 37`)